disable turbolinks on specific links to display chatbox

saipraveen18 opened the pull request:
# PR Checklist

Please check if your PR fulfills the following requirements:
- [X] The title follows our [guidelines](https://github.com/ideacrew/enroll/blob/trunk/CONTRIBUTING.md#commit)
- [ ] Tests for the changes have been added (for bug fixes/features), and they use `let` helpers and `before` blocks.
- [ ] For all UI changes, there is Cucumber coverage.
- [ ] Any endpoint touched in the PR has an appropriate Pundit policy. For open endpoints, the reasoning is documented in the PR and code.
- [ ] Any endpoint modified in the PR only responds to the expected MIME types.
- [ ] For all scripts or rake tasks, how to run them is documented in both the PR and the code.
- [ ] There are no inline styles added.
- [ ] There is no inline JavaScript added.
- [ ] There is no hard-coded text added/updated in helpers/views/JavaScript. New/updated translation strings do not include markup/styles unless there is supporting documentation.
- [ ] Code does not use `.html_safe`.
- [ ] All images added/updated have alt text.
- [ ] Does not bypass RuboCop rules in any way.

# PR Type
What kind of change does this PR introduce?:

- [ ] Bugfix
- [X] Feature (requires Feature flag)
- [ ] Data fix, Migration, or Report (inert code, no impact until run)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Dependency updates (e.g., add a new gem or update to a version)

# What is the ticket # detailing the issue?

Ticket: https://www.pivotaltracker.com/story/show/189022410

# A brief description of the changes:

Current behavior: We are currently using an older version of the Chat feature, while Genesys has upgraded to version 2, which requires us to update our implementation accordingly. Additionally, the chat widget does not appear on certain pages because the new script requires the page to be fully loaded, rather than partially loaded.

New behavior: By disabling Turbolinks on specific pages, we are enabling full page reloads, which in turn triggers the new chat widget to load correctly.

# Feature Flag

For all new feature development, a feature flag is required to control the exposure of the feature to our end users. A feature flag needs a corresponding environment variable to initialize the state of the flag. Please share the name of the environment variable below that would enable/disable the feature and indicate which client(s) it applies to.

Variable name:

GENESYS_CLOUD_MESSENGER_IS_ENABLED - to enable to new Chat feature
GENESYS_CLOUD_MESSENGER_ENV
GENESYS_CLOUD_DEPLOYMENT_ID


- [ ] DC
- [X] ME

# Additional Context
Include any additional context that may be relevant to the peer review process.

saipraveen18 committed:
disable turbolinks on specific links to display chatbox

saipraveen18 committed:
add config in dc yml file

saipraveen18 committed:
fix env in enroll_app.yml file

saipraveen18 committed:
Merge branch 'trunk' into ticket_189022410

saipraveen18 committed:
fix cukes

saipraveen18 committed:
fix rubocop issues

saipraveen18 committed:
Merge branch 'trunk' into ticket_189022410

saipraveen18 committed:
fix chat widget popup issue on enrollments details pages

saipraveen18 committed:
Merge branch 'trunk' into ticket_189022410

saipraveen18 committed:
fix turbolinks issue on plan shoppings page

saipraveen18 committed:
Merge branch 'trunk' into ticket_189022410

saipraveen18 committed:
Merge branch 'trunk' into ticket_189022410

TreyE commented on app/helpers/bs4/bs4_header_helper.rb:
```diff
@@ -11,7 +11,7 @@ def bs4_portal_type(controller) # rubocop:disable Metrics/AbcSize, Metrics/Cyclo
         nil
       elsif current_user.try(:has_hbx_staff_role?)
-        link_to(l10n("layout.header.role.admin"), main_app.exchanges_hbx_profiles_root_path)
+        link_to(l10n("layout.header.role.admin"), main_app.exchanges_hbx_profiles_root_path, data: { turbolinks: false })
```
Disabling turbolinks should be gated behind the same feature flag as the new chat feature.  This is true pretty much everywhere we've decided to set turbolinks: false.

TreyE reviewed (changes requested):
We need to put the disabling of turbo links behind a feature flag.

TristanB17 committed:
Merge branch 'trunk' into ticket_189022410

saipraveen18 committed:
disable turbolinks by feature flag

TristanB17 committed:
Merge branch 'trunk' into ticket_189022410

TristanB17 reviewed (dismissed):
Lgtm!

saipraveen18 committed:
Merge branch 'trunk' into ticket_189022410

TreyE committed:
Recommended CSP changes.

Provided are the recommended changes for the CSP to allow the Genesys
chatbot to load it's assets.  We will need to test this to make sure I
haven't missed any of the endpoints we should be including.

TristanB17 committed:
Merge branch 'trunk' into ticket_189022410
